src/bitcoin-cli.cpp

```diff
@@ -467,8 +467,8 @@ class NetinfoRequestHandler : public BaseRequestHandler
         if (!batch[ID_NETWORKINFO]["error"].isNull()) return batch[ID_NETWORKINFO];
 
         const UniValue& networkinfo{batch[ID_NETWORKINFO]["result"]};
-        if (networkinfo["version"].getInt<int>() < 209900) {
-            throw std::runtime_error("-netinfo requires bitcoind server to be running v0.21.0 and up");
+        if (networkinfo["version"].getInt<int>() < 239900) {
```

Comment:
The commit message appears to have lost the rationale for this bump.

The reason in e6603e50f4fbeeeb5a0dfd25f78b83b85d542984 also doesn't make sense: making a field optional should not break compatibility.

Reply:
(I'd also prefer inlining the rationale by checking the presence of the feature itself, rather than a numeric version comparison...)

Reply:
@luke-jr see #25176 for the patch required that will only be present in the next release.

Reply:
> (I'd also prefer inlining the rationale by checking the presence of the feature itself, rather than a numeric version comparison...)

Multiple fields could change, so it's simpler to check the server version.

Reply:
@jonatack #25176 is backward compatible. Only the server side broke compatibility, and fixing it on the client side shouldn't affect operation with older nodes.

> Multiple fields could change, so it's simpler to check the server version.

Only the first one encountered that would cause an error needs to be checked. This is just a "friendlier error message", right?

Reply:
Ah, you're right. It should instead check for minimum v22 when the 3 `addr_*` fields were added. Your suggestion would be to check if one of them IsNull()? (Yes, a friendlier message, it replaces the "JSON value is not of expected type" errors.)

Reply:
Yes, I'd just remove it temporarily to see where it fails first, and add a check at that location.

Reply:
Done, thanks.

Reply:
Thanks again @luke-jr. I've dropped that last commit and removed "-netinfo" from the PR title and description. After checking back version by version, the -netinfo code always handled the absence or presence of the three addr_* fields in a backward compatible manner, so no need to update -netinfo further for v24.